src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -137,6 +131,31 @@ public void Dispose()
         }
 
         /// <summary>
+        /// Configure the entire service set for this API. Descendants can override this method to choose
+        /// from the default service sets provided by RESTier.
+        /// </summary>
+        /// <param name="apiType">
+        /// CLR type of this <see cref="ApiBase"/>.
+        /// </param>
+        /// <param name="services">
+        /// The <see cref="IServiceCollection"/> with which to create an <see cref="ApiConfiguration"/>.
+        /// </param>
+        /// <returns>The <see cref="IServiceCollection"/>.</returns>
+        [CLSCompliant(false)]
+        protected virtual IServiceCollection ConfigureApiServices(Type apiType, IServiceCollection services)
+        {
+            var customizer = ApiConfiguration.Customize(apiType);
+            services.Apply(customizer.InnerMost);
+
+            return this.ConfigureApi(services)
+                .Apply(customizer.PrivateApi)
+                .UseAttributes(apiType)
+                .UseConventions(apiType)
+                .Apply(customizer.Overrides)
+                .Apply(customizer.OuterMost);
+        }
+
```

Comment:
If we want simplification, IMHO existing configuration options (ConfigureApi) is enough, most users would just use the default configuration to expose a DbContext. The customizer is for advanced scenarios, and whoever uses it must have a knowledge of the default configuration and what each anchor means.
So IMHO we can first make the options fewer and then decide.

Option 1:
We make it extremely simple, only ConfigureApi and ConfigureApiServices (rename them as necessary). And thus we target 2 category of users.
- Most users who want only the basic scenario, no or very little customized configuration, which could be fulfilled by ConfigureApi.
- Some advanced users who use ConfigureApiServices to build completely customized configurations. ConfigureApiServices actually gives user full control on the chain.

Option 2:
We introduce some configuration APIs to ease those advanced users, so that in most cases they don't have to override configuration calls in ApiBase, they just use Customizer.

Option 3:
We go further to support advanced scenarios, allowing configurations created without ApiBase, that's #354